src/solver/opamActionGraph.mli

```diff
@@ -36,16 +36,18 @@ module type SIG = sig
       There is no guarantee however that the resulting graph is acyclic. *)
   val reduce: t -> t
 
-  (** Expand install actions, adding a build action preceding them. *)
-  val explicit: t -> t
-end
+  (** Expand install actions, adding a build action preceding them.
+      The argument [concrete] is a function that should return `true`
+      on package where the `remove` action is a `noop`
+      (such as `conf-*` package). *)
+  val explicit: ?noop_remove:(package -> bool) -> t -> t end
```

Comment:
I am not completely sure this is where this should be done: even if they don't have an effect on the prefix, the installations and removals of `conf` packages get registered in the set of installed packages, and might e.g. have an effect on the environment. As I understand it, a way to express our "optimisation" could be "does not need to be removed before reinstalling", which could be done by simply turning reinstalls to installs beforehand (and making sure that, in case of `conf` packages, installation doesn't fail if the package is already registered as installed).